src/dropdowns/dropdown.ts

```diff
@@ -0,0 +1,136 @@
+import {Directive, Input, Output, EventEmitter, HostBinding, HostListener, ElementRef, OnInit, OnDestroy, ContentChildren, QueryList, Renderer} from 'angular2/core';
+import {NglDropdownItem} from './dropdown-item';
+import {toBoolean} from '../util/util';
+
+const openEventEmitter = new EventEmitter<any>(false);
+
+@Directive({
+  selector: '[nglDropdown]',
+})
+export class NglDropdown implements OnInit, OnDestroy {
+  @Input('open') set isOpen(isOpen: boolean | string) {
+    isOpen = toBoolean(isOpen);
+     if (isOpen) {
+       this._subscribeToPageClickEvents();
+       this.handlePageClickEvents = false;
+       setTimeout(() => this.handlePageClickEvents = true);
+     }
+    this._isOpen = <boolean>isOpen;
+  }
+  get isOpen() {
+    return this._isOpen;
+  }
+  @Input() handlePageEvents = true;
+  @ContentChildren(NglDropdownItem) items: QueryList<NglDropdownItem> = new QueryList<NglDropdownItem>();
+  @Output('openChange') isOpenChange = new EventEmitter<boolean>(false);
+  @HostBinding('class.slds-is-open')
+  @HostBinding('attr.aria-expanded')
+  get __isOpen() {
+    return this.isOpen;
+  }
+
+  private handlePageClickEvents = true;
+  private _isOpen = false;
+  private openEventSubscription: any;
+  private triggerFocusEventEmitter = new EventEmitter(false);
+  private clickEventUnsubscriber: Function = null;
+
+  @HostListener('keydown.esc', ['$event'])
+  @HostListener('keydown.tab')
+  onKeydownClose($event: Event) {
+    this.toggle(false);
+    if ($event) {
```

Comment:
I believe this `if` is a bit tricky here for someone that is reading it :)
You could probably change it to something like: 
```js
@HostListener('keydown.esc', ['$event.keyIdentifier'])
@HostListener('keydown.tab', ['$event.keyIdentifier'])
...
if (key === 'Escape') {
...
}
```